lib/discourse.rb

```diff
@@ -42,6 +42,51 @@ def self.pretty_logs(logs)
       logs.join("\n")
     end
 
+    def self.logs_markdown(logs, user:, filename: 'log.txt')
+      # Reserve 250 characters for the rest of the text
+      max_logs_length = SiteSetting.max_post_length - 250
+      pretty_logs = Discourse::Utils.pretty_logs(logs)
+
+      # If logs are short, try to inline them
+      if pretty_logs.size < max_logs_length
+        return <<~TEXT
+        ```text
+        #{pretty_logs}
+        ```
+        TEXT
+      end
+
+      # Try to create an upload for the logs
+      upload = Dir.mktmpdir do |dir|
+        logfile = File.new(File.join(dir, filename), 'w')
+        logfile.write(pretty_logs)
+        logfile.close
```

Comment:
If you don't need `logfile`, you could maybe use the short version?

```suggestion
        File.write(File.join(dir, filename), pretty_logs)
```